src/oss/langgraph/test.mdx

```diff
@@ -1,3 +1,320 @@
 ---
-title: (Coming Soon) Test
+title: Test
 ---
+
+import AlphaCallout from '/snippets/alpha-lc-callout.mdx';
+
+<AlphaCallout />
+
+After you've prototyped your LangGraph agent, a natural next step is to add tests. This guide covers some useful patterns you can use when writing unit tests.
+
+Note that this guide is LangGraph-specific and covers scenarios around graphs with custom structures - if you are just getting started, check out [this section that uses LangChain prebuilt agents instead](/langchain/test/).
+
+## Prerequisites
+
+:::python
+First, make sure you have [`pytest`](https://docs.pytest.org/) installed:
+
+```bash
+$ pip install -U pytest
+```
+:::
+
+:::js
+First, make sure you have [`vitest`](https://vitest.dev/) installed:
+
+```bash
+$ npm install -D vitest
+```
+:::
+
+## Getting started
+
+Because many LangGraph agents depend on state, a useful pattern is to create your graph before each test where you use it, then compile it within tests with a new checkpointer instance.
+
+The below example shows how this works with a simple, linear graph that progress from `node1` to `node4`. Each node updates a single state key:
+
+:::python
+```python
+import pytest
+
+from typing_extensions import TypedDict
+from langgraph.graph import StateGraph, START, END
+from langgraph.checkpoint.memory import MemorySaver
+
+@pytest.fixture
+def uncompiled_graph():
+    class MyState(TypedDict):
+        my_key: str
+
+    graph = StateGraph(MyState)
+    graph.add_node("node1", lambda state: {"my_key": "hello from node1"})
+    graph.add_node("node2", lambda state: {"my_key": "hello from node2"})
+    graph.add_node("node3", lambda state: {"my_key": "hello from node3"})
+    graph.add_node("node4", lambda state: {"my_key": "hello from node4"})
+    graph.add_edge(START, "node1")
+    graph.add_edge("node1", "node2")
+    graph.add_edge("node2", "node3")
+    graph.add_edge("node3", "node4")
+    graph.add_edge("node4", END)
+    return graph
+
+def test_basic_agent_execution(uncompiled_graph):
+    checkpointer = MemorySaver()
+    compiled_graph = uncompiled_graph.compile(checkpointer=checkpointer)
+    result = compiled_graph.invoke(
+        {"my_key": "initial_value"},
+        config={"configurable": {"thread_id": "1"}}
+    )
+    assert result["my_key"] == "hello from node4"
+```
+:::
+
+:::js
+```ts
+import { test, expect } from 'vitest';
+import {
+  StateGraph,
+  START,
+  END,
+  MemorySaver,
+} from '@langchain/langgraph';
+import { z } from "zod/v4";
```

Comment:
are we using zod/v4 in other places in documentation? could we align on one version so users aren't confused about which version is recommended?